source/documentation/how-to/migrate-pki-to-sha256.html.md

```diff
@@ -0,0 +1,169 @@
+On < 4.1, upgrading to a newer < 4.1 version (e.g. 4.0.6 to 4.0.7) might revert
+this change, so you need to repeat it per each upgrade until 4.1.
+
+On the engine machine, run these commands:
+```sh
+# Backup exiting conf
+cp -p /etc/pki/ovirt-engine/openssl.conf /etc/pki/ovirt-engine/openssl.conf."$(date +"%Y%m%d%H%M%S")"
+
+# Edit it to default to SHA256
+sed -i 's/^default_md = sha1/default_md = sha256/' /etc/pki/ovirt-engine/openssl.conf
+```
+
+## Recreate CA cert
+
+If you only use this procedure because your browser warns/rejects, then it
+might be enough to skip this part. If your browser requires both the CA cert
+and the https cert to have SHA256 signatures, you have to complete it.
+
+On the engine machine, run these commands:
+```sh
+# Backup CA cert
+cp -p /etc/pki/ovirt-engine/private/ca.pem /etc/pki/ovirt-engine/private/ca.pem."$(date +"%Y%m%d%H%M%S")"
+
+# Create a new cert into ca.pem.new
+openssl x509 -signkey /etc/pki/ovirt-engine/private/ca.pem -in /etc/pki/ovirt-engine/ca.pem -out /etc/pki/ovirt-engine/ca.pem.new -days 3650 -sha256
+
+# Replace the existing with the new one
+/bin/mv /etc/pki/ovirt-engine/ca.pem.new /etc/pki/ovirt-engine/ca.pem
+```
+
+## Recreate certs for engine side entities
+
+### Choose entities to recreate
+
+Decide what you want, among the options below:
+
+If only apache httpd (for browsers that reject SHA1 signatures), run:
+
+```sh
+names="apache"
+```
+
+If also the engine cert:
+
+```sh
+names="apache engine"
+```
+
+If all normally-existing entities:
+
+```sh
+names="engine apache websocket-proxy jboss imageio-proxy"
+```
+
+### Enter Maintenance
+
+If this is a self-hosted-engine, move it to global maintenance.
+
+### Recreate
+
+Run this (in the same terminal of previous subsection above):
+
+```sh
+for name in $names; do
+	subject="$(openssl x509 -in /etc/pki/ovirt-engine/certs/"${name}".cer -noout -subject | sed 's;subject= \(.*\);\1;')"
+	/usr/share/ovirt-engine/bin/pki-enroll-pkcs12.sh --name="${name}" --password=mypass --subject="${subject}" --keep-key
+done
+```
+
+### Restart services
+
+If you included apache:
+
+```sh
+systemctl restart httpd
+```
+
+If you included engine:
+
+```sh
+systemctl restart ovirt-engine
+```
+
+If you included ovirt-websocket-proxy/ovirt-imageio-proxy:
+
+```sh
+systemctl restart ovirt-websocket-proxy
+
+systemctl restart ovirt-imageio-proxy
+```
+
+### Exit maintenance
+
+If this is a self-hosted-engine, exit global maintenance.
+
+### Reconnect to web admin
+
+Your browser will likely refuse to continue working with the web admin ui.
+You might need to restart it and/or remove the engine cert and/or engine ca cert.
+
+In my own case I unchecked "Permanently store this exception" when I first
+logged in, and after restarting httpd the browser showed an error about using
+the same serial number. Restarting the browser was enough to login again.
+
+## Re-add hosts
+
+For all of your hosts, one host at a time, using the web admin ui:
+* Set it to Maintenance
```

Comment:
How can you set it to maint.? If you've recreated the CA, how does it communicate with the hosts?